src/Compilers/CSharp/Portable/Binder/Binder_AnonymousTypes.cs

```diff
@@ -192,19 +192,7 @@ private TypeSymbol GetAnonymousTypeFieldType(BoundExpression expression, CSharpS
                 }
                 else
                 {
-                    if (expression.Kind == BoundKind.UnboundLambda)
-                    {
-                        errorArg = ((UnboundLambda)expression).MessageID.Localize();
-                    }
-                    else if (expression.Kind == BoundKind.MethodGroup)
-                    {
-                        errorArg = MessageID.IDS_MethodGroup.Localize();
-                    }
-                    else
-                    {
-                        Debug.Assert(expression.IsLiteralNull(), "How did we successfully bind an expression without a type?");
-                        errorArg = MessageID.IDS_NULL.Localize();
-                    }
+                    errorArg = expression.Display;
```

Comment:
Is this change related to the feature work? If not, we might want to put it in the 'future' on its own. Please create an issue for this.